vm/src/builtins/bytes.rs

```diff
@@ -128,6 +128,15 @@ impl PyBytes {
         &self.inner.elements
     }
 
+    #[pymethod(name = "__bytes__")]
+    pub fn bytes(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyObjectRef {
+        if zelf.is(&vm.ctx.types.bytes_type) {
+            zelf.into_object()
+        } else {
+            PyBytes::from(zelf.inner.elements.to_vec()).into_object(vm)
+        }
+    }
```

Comment:
Don't think you can avoid the clone here.

```suggestion
    #[pymethod(magic)]
    fn bytes(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyRef<PyBytes> {
        if zelf.is(&vm.ctx.types.bytes_type) {
            zelf
        } else {
            PyBytes::from(zelf.inner.clone()).into_ref(vm)
        }
    }
```

Reply:
You are right. It seems extremely better than original. Thanks. I sent v3 code.